### PR TITLE
Need to maintain consistent .proto message field numbers

### PR DESCRIPTION
**Pull request by alakendu**

### CODE REVIEW

**Review by Rangababu-R — CHANGES_REQUESTED** [review]

**Comment by Rangababu-R** [comment]
Can't the enum be below format? why do we need to add x-enum?
```yaml
enum:
  - status_200:
    x-field-uid: 1
  - status_400:
    x-field-uid: 2
  - status_404:
    x-field-uid: 3
  - status_500:
    x-field-uid: 4
```

File: openapiart/openapiartprotobuf.py [hunk 1]
```diff
@@ -339,14 +365,31 @@ def _write_response_fields(self, msg_name, schema_object):
...
+            )
+
     def _write_msg_fields(self, name, schema_object):
         if "properties" not in schema_object:
             return
-        id = 0
+        field_uids = []
+        reserved_field_uids = []
+        if "x-reserved-field-uids" in schema_object:
+            reserved_field_uids = schema_object["x-reserved-field-uids"]
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/openapiartprotobuf.py]: done

File: openapiart/openapiartprotobuf.py [hunk 2]
```diff
@@ -339,14 +365,31 @@ def _write_response_fields(self, msg_name, schema_object):
         except AttributeError as err:
             print("Failed writing response {}: {}".format(msg_name, err))
 
+    def _check_duplicate_uid(self, fields_uid, name):
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/openapiartprotobuf.py]: done

File: openapiart/openapiartprotobuf.py [hunk 3]
```diff
@@ -295,22 +303,40 @@ def _get_description(self, openapi_object):
...
         if "unspecified" not in enums:
-            enums.insert(0, "unspecified")
-        id = 0
-        for enum in enums:
-            self._write("{} = {};".format(enum.lower(), id), indent=3)
-            id += 1
+            self._write("{} = {};".format("unspecified", 0), indent=3)
+        field_uids = []
+        for key, value in enums.items():
+            if "x-field-uid" not in value:
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/openapiartprotobuf.py]: done

File: utils/autofielduid.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
+import os
```
[the remark this answers is not in the document]
Comment (alakendu) [reply utils/autofielduid.py]: ok

File: openapiart/bundler.py [hunk 5]
```diff
@@ -105,6 +118,39 @@ def _validate_errors(self):
...
+            "schemas"
+        ].items():
+            if "properties" not in schema_object:
+                continue
+            for property_name, property_object in schema_object[
+                "properties"
+            ].items():
+                if "type" not in property_object.keys():
+                    continue
+                if property_object["type"] == "string":
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/bundler.py:144]: I think our latest opeanapiart are covering only for string. Please let me know if I have missed something.

File: openapiart/bundler.py [hunk 6]
```diff
@@ -219,13 +265,31 @@ def _resolve_refs(self, base_dir, yobject):
                     inline = self._get_inline_ref(base_dir, refs[0], refs[1])
                     yobject[key] = inline
                 elif key == "x-include":
-                    for include_ref in value:
-                        if include_ref not in self._includes:
-                            include = self._get_schema_object(
+                    if not isinstance(value, str):
+                        self._errors.append(
+                            "x-include should be part of properties and responses for %s"
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/bundler.py:378]: modify it

File: openapiart/bundler.py [hunk 7]
```diff
@@ -219,13 +265,31 @@ def _resolve_refs(self, base_dir, yobject):
...
+                            % value
+                        )
+                        continue
+                    if value not in self._includes:
+                        file_name, include_path = value.split("#")
+                        paths = include_path.split("/")
+                        obj_path = "/".join(paths[:4])
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/bundler.py]: done

File: openapiart/bundler.py [hunk 8]
```diff
@@ -219,13 +265,31 @@ def _resolve_refs(self, base_dir, yobject):
...
+                        if include_ref in self._include_objects:
+                            include_object = self._include_objects[include_ref]
+                        else:
+                            include_object = self._get_schema_object(
                                 base_dir, include_ref
                             )
-                            self._resolve_refs(base_dir, include)
-                            self._includes[include_ref] = include
+                            self._include_objects[include_ref] = include_object
+                        for node in paths[4:]:
```
[the remark this answers is not in the document]
Comment (alakendu) [reply openapiart/bundler.py]: done

**Comment by ankur-sheth** [comment]
I'm not the right person to review this.  As long as the team has reviewed it, I'm ok.

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**